12_6_Accessing_Bitcoind_with_Python.md

```diff
@@ -0,0 +1,86 @@
+# 12.6: Accessing Bitcoind with Python
+
+> **NOTE:** This is a draft in progress, so that I can get some feedback from early reviewers. It is not yet ready for learning.
+
+Though command-line `curl` is the easiest way to access the `bitcoind` directly, there are many other options for doing so and most of them support more fully featured programming languages. One package for doing so in Python is [Bitcoin authproxy](https://github.com/bitcoin/bitcoin/blob/master/test/functional/test_framework/authproxy.py). It was originally [written by Jeff Garzik](https://github.com/jgarzik/python-bitcoinrpc) but is now maintained by Bitcoin Core as the primary interface for the QA testing suite. It uses an http client module to access the `bitcoind` RPC server.
+
+## Set Up Bitcoin AuthProxy
+
+The latest Bitcoin Core version of `authproxy` requires Python 3, but you can still use jgarzik's original script to run in Python 2. Download the script to your working directory, start Python, and import the module. This example uses Python 3 running on Linux:
+```
+$ python3
+>>> from authproxy import AuthServiceProxy, JSONRPCException
+```
+Instantiate an `AuthServiceProxy` object with the same localhost IP address, port, username and password you set in `bitcoin.conf`. For exmaple if these are the lines in `bitcoin.conf`:
+```
+rpcuser=bitcoinrpc
+rpcpassword=1d446e2e45e96aaba6dea515871cc497
+```
+your Python command would be:
+```
+>>> rpc_connection = AuthServiceProxy("http://%s:%s@127.0.0.1:8332"%("bitcoinrpc","1d446e2e45e96aaba6dea515871cc497"))
+
```

Comment:
I assume this is part of the Python code file? It'd be great to show more of the context, and we also want to have the complete code files at the end.